tuist/tuist
# Add support for watchOS extension to have WidgetKit extension dependencies

### Short description 📝

- WatchOS extension targets can now include WidgetKit extension targets (as of Xcode 14)
- WidgetKit extension targets are regular app extension targets
- Tuist already supports embedding app extensions, the only change needed to support it for watchOS extension was to update the linter logic to allow this dependency combination
- The fixture was extended to demo a watchOS WidgetKit extension

Notes:

- WidgetKit extension targets require the following info plist keys
```
<dict>
	<key>NSExtension</key>
	<dict>
		<key>NSExtensionPointIdentifier</key>
		<string>com.apple.widgetkit-extension</string>
	</dict>
</dict>
```

### How to test the changes locally 🧐

- Generate the watch application fixture

```
swift build
swift run tuist generate --path projects/tuist/fixtures/ios_app_with_watchapp2
```

- Verify the project generates successfully without any lint errors
- Inspect the generated project and verify the WidgetKit extension for the watch extension is created correctly

### Contributor checklist ✅


### Reviewer checklist ✅
